### ISSUE TITLE
Blade components from package not recognised

### ISSUE THREAD

**Issue by dsazup** [laravel-idea/plugin#285]
Hello, I'm using this really cool form package and the components are registered in this way: https://github.com/protonemedia/laravel-form-components/blob/master/src/Support/ServiceProvider.php#L52

Wondering if there is any chance the plugin could recognise it as blade components and "go to declaration" would be working?

**Comment by adelf** [comment]
Hi, Dainius.
Yes, there are some packages with this kind of registration. I'm going to make it work. Thanks for another example)

**Comment by mertasan** [comment]
I need this too. +1

<img width="741" alt="screenshot 19" src="https://user-images.githubusercontent.com/13007665/121176701-3288a480-c865-11eb-9797-9a672359a936.png">
<img width="526" alt="screenshot 18" src="https://user-images.githubusercontent.com/13007665/121176713-35839500-c865-11eb-818d-66bada48b6d0.png">

**Comment by adelf** [comment]
@mertasan it will be implemented by configuring ide.json file. I hope in the next version.

**Comment by adelf** [comment]
@mertasan I remember) Sorry. I want to make version 2 and rewrite the docs after that. I'll remove this text)

**Comment by natepisarski** [comment]
@adelf I'm sorry for the ping, but just to clarify something from this comment:
> @mertasan it will be implemented by configuring ide.json file. I hope in the next version.

is this sort of thing possible today, with some form of `ide.json` configuration? Or will a future version of the plugin add this functionality (which will then be configured via `ide.json`)?

If it's possible today, could you share an example of what it would look like to get components from the `project-x` prefix working with the plugin? I really need this kind of functionality, and I've tried numerous configurations in `ide.json` to no avail.

In my particular case, there is no php Namespace that is unique to the package - it's simply `App`. The same as the Laravel projects which consume it. But the components are properly registered / usable. So I don't know if that's the reason I can't get the completion to work.

**Comment by adelf** [comment]
Could you share your package or the component registration code?

**Comment by mertasan** [comment]
The schema I use for my project. Maybe it helps. @natepisarski 

```
{
    "$schema": "http://laravel-ide.com/schema/laravel-ide-v1.json",
    "blade": {
        "components": {
            "viewPaths": [
                {
                    "path": "resources/views/includes",
                    "namespace": "tdt-includes"
                },
                {
                    "path": "resources/views/components",
                    "namespace": "tdt-components"
                },
                {
                    "path": "resources/views/components",
                    "namespace": "",
                    "prefix": "tdt-"
                }
            ]
        }
    }
}
```

```blade
@include('tdt-includes::files.file')

@include('tdt-components::files.file')

<x-tdt-files.file :hello="world" />

```

**Comment by natepisarski** [comment]
So, in this scenario I can't share the real example since it's company code. But an example of what I'm doing is this:

* Have a Component class in `app\View\Components\...\MyFile.php` where `...` is some arbitrary path.
* Have a view in `resources\views\Components\...\my-file.blade.php` (`...` is some arbitrary path again, usually matching the Component file.)
* Have a service provider that does the following:

```php
 $this->loadViewComponentsAs('project-x', [
    'my-file' => App\View\Components\...\MyFile::class
]);
```

And finally, in the client project which consumes this package:

```php
<x-project-x-my-file thing="a" />
```

And in this scenario, `<x-project-x-my-file>` is highlighted with the `Unknown Blade component` error. And has no jump-to-definition, or prop auto-completion, etc.

---

@mertasan Thank you, I'm going to try working from this example and report back. I think when I was playing with the `ide.json` I was misinterpreting what the "namespace" property was for.

**Comment by adelf** [comment]
Thanks, Nate. I’ll check this tomorrow.

**Comment by natepisarski** [comment]
@mertasan To follow up, what you posted led to me being able to get this working! I think the exact problem here was that I use some pretty deeply nested folders. I also misunderstood exactly what `prefix` was. So for a more full example if I had this:

* Component class is `app\View\Components\Inputs\Text.php`
* Resource is `resources/views/components/inputs/text.blade.php`
* Registration is

```php
$this->loadViewComponentsAs('project-x', [
  'inputs.text' => App\View\Components\Inputs\Text::class,
]);
$this->publishes('/../../resources/views/components', resource_path('views/vendor/project-x/components);
```

Then the `ide.json` that makes it work will be

```json
        {
          "path": "resources/views/vendor/project-x/components/input",
          "prefix": "project-x-input"
        }
```

Now this auto-completes props:

```
<x-project-x-inputs.text 
```

So basically, per nested folder, I'll need to add an entry into `ide.json` which I'm fine with. Before I was just under the impression that

```json
        {
          "path": "resources/views/vendor/project-x/components",
          "prefix": "project-x"
        }
```

would work, and automatically figure out what was going on in nested directories / registrations.

So, in a sense I guess there is a "problem" here, in that the plugin can't automatically detect things like this. But the configuration in `ide.json` is working 100% as intended.

**Comment by wvdongen** [comment]
@adelf Any updates on the initial question? Can you please describe how to set up `ide.json` in order to properly recognize https://github.com/protonemedia/laravel-form-components?

**Comment by wvdongen** [comment]
Something like this seems to work:

```json
{
    "$schema": "http://laravel-ide.com/schema/laravel-ide-v1.json",
    "blade": {
        "components": {
            "viewPaths": [
                {
                    "path": "vendor/protonemedia/laravel-form-components/resources/views/bootstrap-4"
                },
                {
                    "path": "vendor/protonemedia/laravel-form-components/src/Components",
                    "prefix": "form-"
                }
            ]
        }
    }
}
```

**Comment by adelf** [comment]
@wvdongen this doesn't match this package components registration. I have to parse config files to register them. I'll implement them... probably in the next version.

**Comment by adelf** [comment]
https://github.com/protonemedia/laravel-form-components/pull/81

Laravel Idea 5.0 is out. PR is created by me. I think I can close this :)

**Comment by dansleboby** [comment]
Should be reopen, 

@wvdongen code seems fine just a bit update to make it work inside the plugin itself

```json
{
    "$schema": "http://laravel-ide.com/schema/laravel-ide-v2.json",
    "blade": {
        "components": {
            "viewPaths": [
                {
                    "path": "resources/views/bootstrap-4"
                },
                {
                    "path": "src/Components",
                    "prefix": "form-"
                }
            ]
        }
    }
}
```

I think a note should be add in the readme about the incompatibility of IDE and custom prefix, of make a command that gen the right one

But thanks @wvdongen for help!
